src/libraries/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -1862,49 +1642,19 @@ private void HandleDisconnect()
 
                 if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, $"DisconnectResults {listener.DisconnectResults} removing for _connectionId: {_connectionId}");
                 listener.DisconnectResults.Remove(_connectionId);
-                _session?.CloseContext();
 
-                // Clean up the identity. This is for scenarios where identity was not cleaned up before due to
-                // identity caching for unsafe ntlm authentication
-
-                IDisposable? identity = _authenticatedConnection == null ? null : _authenticatedConnection.Identity as IDisposable;
-                if ((identity != null) &&
-                    (_authenticatedConnection!.Identity.AuthenticationType == AuthenticationTypes.NTLM) &&
-                    (listener.UnsafeConnectionNtlmAuthentication))
-                {
-                    identity.Dispose();
-                }
```

Comment:
This removal is likely wrong. The identity is now cloned (late change to match ASP.NET, since claims can be changed on the identity and it's therefore mutable) and it has to be disposed.